### PR TITLE
Add CVar to change the MP5 fire-rate (sv_ag_mp5_old_rate)

### PR DESCRIPTION
**Pull request by sabianroberts**
In the WON versions of Half-Life, as proven by [this video at timestamp 2:20,](https://www.youtube.com/watch?v=uKkJPFmDW9Q) the fire-rate for the MP5 was slightly faster than it is in the current Steam version of HL, including AG. I thought it would be nice to add this CVar for version 6.7, as players may want the old fire-rate.

### CODE REVIEW

**Comment by SmileyAG** [comment]
Firing rate in GoldSrc was depends from FPS: https://wiki.sourceruns.org/wiki/FPS_Effects#Firing_rate

WON version initially capped to 72 FPS (**fps_single**), that means we should cap Steam version to 72 FPS too and make comparison, since default **fps_max** was 100 for Steam.

Comparison: 

https://user-images.githubusercontent.com/58108407/122649430-e2e88980-d13e-11eb-8d86-6bc194f4b99d.mp4

You also could make sure that the value for **m_flNextPrimaryAttack** is **0.1** for both versions,

HLSDK 1.0: https://github.com/a1batross/HLSDK_ancient/blob/master/DLLS/MP5.CPP#L215
Steampipe: https://github.com/ValveSoftware/halflife/blob/master/dlls/mp5.cpp#L191

Still not enough proofs? Then we can decompile both **hl.dll** in https://github.com/NationalSecurityAgency/ghidra and make sure the value is **0.1**:

ghidra-won

ghidra-steampipe

So, basically, you could reach the fastest firing rate by experiments with FPS.

**Comment by fireblizzard** [comment]
Damn Smiley you beat me to it :) Thanks for your reply.

I'm not against the idea as long as it's true that the MP5 in WON had a slower firerate by itself (not affected by other settings, like FPS). But i'm not sure about that. That video is not enough proof, as it doesn't even show FPS. We need to test in WON itself, and it's been a very long time since i last played WON so i would appreciate if someone else that is more familiarized could do it :D
(I think Smiley just did this test in his comparison video)

Then as Smiley has pointed out, I haven't seen any HLSDK version that changed anything related to MP5's m_flNextPrimaryAttack, it seems that it's always been assigned the same float literals (0.1 in this case).

I'm just gonna leave some resources so you can do further research. Dates for HL versions, different HLSDKs, and the research about FPS and firerate.

HL changelog: http://hl.loess.ru/?txt=changelog1110

Old HLSDKs:
* http://metamod.org/files/sdk/old/
* https://web.archive.org/web/19991002235638/http://ftp.won.net/pub/half-life/sdk/
* https://github.com/a1batross/HLSDK_ancient

Firerate research:
* http://hl.loess.ru/?txt=fps_dependence
* http://www.fortress-forever.com/fpsreport/

image

**Comment by SmileyAG** [comment]
> We need to test in WON itself, and it's been a very long time since i last played WON so i would appreciate if someone else that is more familiarized could do it :D
> (I think Smiley just did this test in his comparison video)

Yeah that exactly what I done, in my comparison you could also notice **cl_showfps 1** for Steampipe version

Reason, why I haven't done fps encounter for WON, ~~cuz it requires using 3rd-party tools (Shadow Play or MSI Afterburner e.g.)~~, I didn't found cvar like **cl_showfps** or **net_graph** for WON version, maybe someone knows it.

UPD: I found, **r_netgraph 1**, so I can redo the comparison if needed.

**Comment by sabianroberts** [comment]
> Firing rate in GoldSrc was depends from FPS: https://wiki.sourceruns.org/wiki/FPS_Effects#Firing_rate
> 
> WON version initially capped to 72 FPS (**fps_single**), that means we should cap Steam version to 72 FPS too and make comparison, since default **fps_max** was 100 for Steam.

Oh shite, I didn't know that. Thanks for letting me know dude! Looks like I have to update HL Classic now... lol

But does anyone think the CVar could still be useful for some matches/gamemodes? Some players & server operators might like the feature :)

**Comment by fireblizzard** [comment]
I think the MP5 is fine as it is, 0.09 seems like too overpowered, from my experience. Anyways, in this case i would only accept a PR that adds the ability to change the firerate of every weapon, i don't know why we would only want it for the MP5. And it might be a bit of a waste of time, since i'm looking forward to Solokiller's Enhanced Half-Life, whose changes i'll try to apply to AG and will probably replace whatever work we do in this regard.

I'm also gonna point out things that i see wrong in the commit, just for the sake of correctness, even if this not mergeable for the reasons explained.
